numpy/lib/function_base.py

```diff
@@ -1526,20 +1541,42 @@ def unwrap(p, discont=pi, axis=-1):
     array([ 0.        ,  0.78539816,  1.57079633,  5.49778714,  6.28318531]) # may vary
     >>> np.unwrap(phase)
     array([ 0.        ,  0.78539816,  1.57079633, -0.78539816,  0.        ]) # may vary
-
+    >>> unwrap([0, 1, 2, -1, 0], interval_size=4)
+    array([0., 1., 2., 3., 4.])
+    >>> unwrap([ 1, 2, 3, 4, 5, 6, 1, 2, 3], interval_size=6)
+    array([1., 2., 3., 4., 5., 6., 7., 8., 9.])
+    >>> unwrap([2, 3, 4, 5, 2, 3, 4, 5], interval_size=4)
+    array([2., 3., 4., 5., 6., 7., 8., 9.])
+    >>> phase_deg = np.mod(np.linspace(0,720,19), 360) - 180
+    >>> unwrap(phase_deg, interval_size=360)
+    array([-180., -140., -100.,  -60.,  -20.,   20.,   60.,  100.,  140.,
+        180.,  220.,  260.,  300.,  340.,  380.,  420.,  460.,  500.,
+        540.])
     """
     p = asarray(p)
     nd = p.ndim
     dd = diff(p, axis=axis)
+    offset = 0
+    if (not min_val is None) and (not max_val is None):
+        interval_size = max_val - min_val
+        slice0list = [slice(None, None)]*nd     # full slices
+        slice0list[axis] = 0
+        slice0 = tuple(slice0list)
+        offset_mul = (p[slice0] - min_val)//interval_size
+        slice0list[axis] = None
+        offset = -offset_mul[tuple(slice0list)]*interval_size
```

Comment:
I think this becomes easier as:
```suggestion
        slice0list[axis] = slice(0, 1)
        slice0 = tuple(slice0list)
        offset_mul = ((p[slice0] - min_val)//interval_size)*interval_size
```

Reply:
I'm not so familiar with this way of slicing yet. So if I got it right, than `offset_mul` will have the same dimensions as `p`, but not the same shape. With my solution it would not even have the same dimension, right? But I think a minus is missing

Reply:
Yeah, likely I screwed that snippet up. The key thing is to index with `0:1` instead of `0`, which avoids the need to use `None`.